pkg/monitor/monitor/monitor_manager.go

```diff
@@ -427,3 +471,186 @@ func (m *MonitorManager) patchTidbClusterStatus(tcRef *v1alpha1.TidbClusterRef,
+						return false, err
+					}
+					//monitor patch status successfully
+				}
+
+				err = m.deps.PVCControl.DeletePVC(monitor, &corev1.PersistentVolumeClaim{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      deploymentPvcName,
+						Namespace: monitor.Namespace,
+					},
+				})
+				if err != nil {
+					klog.Errorf("Fail to delete the PVC %s for tm [%s/%s], err: %v", deploymentPvcName, monitor.Namespace, monitor.Name, err)
+					return false, err
+				}
+
+				deploymentPv, err := m.deps.PVLister.Get(deploymentPvc.Spec.VolumeName)
+				if err != nil && !errors.IsNotFound(err) {
+					klog.Errorf("Smooth migration for tm[%s/%s], fail to get PV %s, err: %v", monitor.Namespace, monitor.Name, deploymentPvc.Spec.VolumeName, err)
+					return false, err
+				}
+
+				if deploymentPv.Spec.ClaimRef != nil && deploymentPv.Spec.ClaimRef.Name == firstStsPvcName {
+					// smooth migration successfully and clean status
+					monitor.Status.DeploymentStorageStatus = nil
+					return true, nil
+				}
+
+				if deploymentPv.Spec.ClaimRef == nil {
+					deploymentPv.Spec.ClaimRef = &corev1.ObjectReference{}
+				}
+
+				deploymentPv.Spec.ClaimRef.Name = firstStsPvcName
+				err = m.deps.PVControl.PatchPVClaimRef(monitor, deploymentPv, firstStsPvcName)
+				if err != nil {
+					klog.Errorf("Smooth migration for tm[%s/%s], fail to patch PV %s, err: %v", monitor.Namespace, monitor.Name, deploymentPv.Name, err)
+					return false, err
+				}
+				// smooth migration successfully and clean status
+				monitor.Status.DeploymentStorageStatus = nil
+				return true, nil
+			}
+			return true, nil
+		} else {
+			klog.Errorf("Fail to get deployment for tm [%s/%s], err: %v", monitor.Namespace, monitor.Name, err)
+			return false, err
+		}
+	}
+	klog.Infof("The old deployment exists, start smooth migration for tm [%s/%s]", monitor.Namespace, monitor.Name)
+	// if deployment exist, delete it and wait next reconcile.
+	err = m.deps.TypedControl.Delete(monitor, &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      oldDeployment.Name,
+			Namespace: oldDeployment.Namespace,
+		},
+	})
+	if err != nil {
+		klog.Errorf("smoothMigration tm[%s/%s]'s,old deployment failed to delete,err: %v", monitor.Namespace, monitor.Name, err)
+		return false, err
+	}
+	// If enable persistent,operator need to migrate pvc and pv binding relationship.
+	return !monitor.Spec.Persistent, nil
+
+}
+
+func (c *MonitorManager) validate(tidbmonitor *v1alpha1.TidbMonitor) bool {
+	errs := v1alpha1validation.ValidateTidbMonitor(tidbmonitor)
+	if len(errs) > 0 {
+		aggregatedErr := errs.ToAggregate()
+		klog.Errorf("tidbmonitor %s/%s is not valid and must be fixed first, aggregated error: %v", tidbmonitor.GetNamespace(), tidbmonitor.GetName(), aggregatedErr)
+		c.recorder.Event(tidbmonitor, corev1.EventTypeWarning, "FailedValidation", aggregatedErr.Error())
+		return false
+	}
+	return true
+}
+
+func (m *MonitorManager) syncTidbMonitorPV(tm *v1alpha1.TidbMonitor) error {
```

Comment:
This seems to duplicate with the `Sync` in `meta_manager.go`, I think we can change the parameter of `Sync` in `meta_manager.go` to handle different cases.